src/Tasks/GenerateResource.cs

```diff
@@ -3387,8 +3393,13 @@ private bool HaveSystemResourcesExtensionsReference
 
                 PopulateAssemblyNames();
 
-                foreach (var assemblyName in _assemblyNames)
+                foreach (AssemblyNameExtension assemblyName in _assemblyNames)
                 {
+                    if (assemblyName == null)
```

Comment:
Why would they be null?

Reply:
That's what happens if an assembly name can't be extracted from a reference for some reason in `PopulateAssemblyNames()`. The other codepaths were robust to this situation but this newly-added one wasn't (until running a test revealed the error).